Windows10DebloaterGUIOLD.ps1

```diff
@@ -172,82 +341,17 @@ Start-Transcript -OutputDirectory "$DebloatFolder"
-                "Microsoft.Microsoft3DViewer"
-                "Microsoft.MicrosoftOfficeHub"
-                "Microsoft.MicrosoftSolitaireCollection"
-                "Microsoft.NetworkSpeedTest"
-                "Microsoft.News"
-                "Microsoft.Office.Lens"
-                "Microsoft.Office.OneNote"
-                "Microsoft.Office.Sway"
-                "Microsoft.OneConnect"
-                "Microsoft.People"
-                "Microsoft.Print3D"
-                "Microsoft.RemoteDesktop"
-                "Microsoft.SkypeApp"
-                "Microsoft.StorePurchaseApp"
-                "Microsoft.Office.Todo.List"
-                "Microsoft.Whiteboard"
-                "Microsoft.WindowsAlarms"
-                #"Microsoft.WindowsCamera"
-                "microsoft.windowscommunicationsapps"
-                "Microsoft.WindowsFeedbackHub"
-                "Microsoft.WindowsMaps"
-                "Microsoft.WindowsSoundRecorder"
-                "Microsoft.Xbox.TCUI"
-                "Microsoft.XboxApp"
-                "Microsoft.XboxGameOverlay"
-                "Microsoft.XboxIdentityProvider"
-                "Microsoft.XboxSpeechToTextOverlay"
-                "Microsoft.ZuneMusic"
-                "Microsoft.ZuneVideo"
-
-                #Sponsored Windows 10 AppX Apps
-                #Add sponsored/featured apps to remove in the "*AppName*" format
-                "*EclipseManager*"
-                "*ActiproSoftwareLLC*"
-                "*AdobeSystemsIncorporated.AdobePhotoshopExpress*"
-                "*Duolingo-LearnLanguagesforFree*"
-                "*PandoraMediaInc*"
-                "*CandyCrush*"
-                "*Wunderlist*"
-                "*Flipboard*"
-                "*Twitter*"
-                "*Facebook*"
-                "*Spotify*"
-                "*Minecraft*"
-                "*Royal Revolt*"
-                "*Sway*"
-                "*Dolby*"
-                "*Windows.CBSPreview*"
-                
-                #Optional: Typically not removed but you can if you need to for some reason
-                #"*Microsoft.Advertising.Xaml_10.1712.5.0_x64__8wekyb3d8bbwe*"
-                #"*Microsoft.Advertising.Xaml_10.1712.5.0_x86__8wekyb3d8bbwe*"
-                #"*Microsoft.BingWeather*"
-                #"*Microsoft.MSPaint*"
-                #"*Microsoft.MicrosoftStickyNotes*"
-                #"*Microsoft.Windows.Photos*"
-                #"*Microsoft.WindowsCalculator*"
-                #"*Microsoft.WindowsStore*"
-            )
-            foreach ($Bloat in $Bloatware) {
-                Get-AppxPackage -Name $Bloat| Remove-AppxPackage
-                Get-AppxProvisionedPackage -Online | Where-Object DisplayName -like $Bloat | Remove-AppxProvisionedPackage -Online
-                Write-Host "Trying to remove $Bloat."
-                Write-Host "Bloatware removed!"
+            foreach ($Bloat in $global:Bloatware) {
+                if ( !($NonRemovables -cmatch $Bloat) ) {
```

Comment:
On second thought, I've given the tools needed to avoid/detect conflicts. If the user wants to specifically add an item to the blacklist even though the computer reports that it is non-removable, we should still honour the request to remove it.  What if the computer is lying, and the person is smarter?